test/compare_sgemm_sbgemm.c

```diff
@@ -81,6 +81,16 @@ float16to32 (bfloat16_bits f16)
   return f32.v;
 }
 
+float
+float32to16 (float32_bits f32)
```

Comment:
float32_bits is not defined in some cases.

```
compare_sgemm_sbgemm.c:121:39: error: passing 'float' to parameter of incompatible type 'float32_bits'
  121 |               AA[j * k + i].v = float32to16( A[j * k + i] );
      |                                              ^~~~~~~~~~~~
compare_sgemm_sbgemm.c:85:27: note: passing argument to parameter 'f32' here
   85 | float32to16 (float32_bits f32)
      |                           ^
compare_sgemm_sbgemm.c:122:39: error: passing 'float' to parameter of incompatible type 'float32_bits'
  122 |               BB[j * k + i].v = float32to16( B[j * k + i] );
      |                                              ^~~~~~~~~~~~
compare_sgemm_sbgemm.c:85:27: note: passing argument to parameter 'f32' here
   85 | float32to16 (float32_bits f32)
```

Reply:
strange, this is typedef'd as a union near the top of the file (same as with bfloat16_bits)

Reply:
Sorry, it is defined.  It just is NOT compatible.

Reply:
Your changes also break some of the unit tests.

Reply:
which would that be ?

Reply:
BTW, converting from float32 to bfloat16 is not as simple as shifting bits.  You need to take rounding into account (plus more if you are considering denormalization, etc)

Reply:
```
OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./test_sbgemm > SBBLAT3.SUMM
FATAL ERROR SBGEMM - Return code: 338349
```

Reply:
not sure why I did not see that... will fix in a followup PR

Reply:
#4491